infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/StreamsSnapshotWriter.java

```diff
@@ -371,8 +423,23 @@ private void applyShadowStream(UUID streamId, long snapshot) {
                 txnContext.commit();
             }
         }
+
+        try (TxnContext txn = metadataManager.getTxnContext()) {
+            ReplicationMetadata metadata = metadataManager.queryReplicationMetadata(txn, session);
+            if (metadata.getTopologyConfigId() != topologyConfigId || metadata.getLastSnapshotStarted() != srcGlobalSnapshot) {
+                log.warn("Metadata mismatch detected.  In-memory topologyId/lastSnapshotStarted = {}/{}.  From " +
+                        "metadata table = {}/{}", topologyConfigId, srcGlobalSnapshot, metadata.getTopologyConfigId(),
+                        metadata.getLastSnapshotStarted());
+                throw new ReplicationWriterException("Metadata mismatch detected");
+            }
+
+            ReplicationMetadata.Builder metadataBuilder = metadata.toBuilder();
+            metadataBuilder.setLastSnapshotStarted(srcGlobalSnapshot);
+            metadataManager.updateReplicationMetadata(txn, session, metadataBuilder.build());
+            txn.commit();
```

Comment:
can we capture this commit timestamp and also include it in the log statement below just to help with debugging?